### PR TITLE
DPP-368 clean up flags

### PR DESCRIPTION
**Pull request by rautenrieth-da**
Changes:
- Make sure all parameters of the parallel indexer can be configured through the CLI (in ledger-on-x, not in sandbox)
- Fail if incompatible CLI flags are used

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala [hunk 1]
```diff
@@ -146,11 +146,12 @@ object JdbcIndexer {
       for {
         // Note: the LedgerDao interface is only used for initialization here, it can be released immediately
         // after initialization is finished. Hence the use of ResourceOwner.use().
+        // TODO append-only: Rewrite the initialization using the StorageBackend interface, and remove the use of LedgerDao from this class
         _ <- com.daml.platform.store.appendonlydao.JdbcLedgerDao
           .writeOwner(
             serverRole,
             config.jdbcUrl,
-            config.databaseConnectionPoolSize,
+            connectionPoolSize = 1,
```
Comment (rautenrieth-da) [inline ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala:154]: `LedgerDao` is only used for sequential initialization here, 1 connection at a time is enough

File: ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala [hunk 2]
```diff
@@ -146,11 +146,12 @@ object JdbcIndexer {
       for {
         // Note: the LedgerDao interface is only used for initialization here, it can be released immediately
         // after initialization is finished. Hence the use of ResourceOwner.use().
+        // TODO append-only: Rewrite the initialization using the StorageBackend interface, and remove the use of LedgerDao from this class
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala:149]: this I accidentally already did in another PR, you can as good as refrain changing this file (one less conflict to worry about)

File: ledger/participant-integration-api/src/main/scala/platform/indexer/IndexerConfig.scala [hunk 3]
```diff
@@ -15,6 +15,7 @@ case class IndexerConfig(
     participantId: Ref.ParticipantId,
     jdbcUrl: String,
     startupMode: IndexerStartupMode,
+    // TODO append-only: remove after removing support for the current (mutating) schema
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/main/scala/platform/indexer/IndexerConfig.scala:18]: why did you remove the TODO from here?

**Review by nmarton-da — APPROVED** [review]
LGTM

**Review by mziolekda — APPROVED** [review]
LGTM

File: ledger/participant-integration-api/src/main/scala/platform/indexer/IndexerConfig.scala (hunk 3 above)
Comment (mziolekda) [reply ledger/participant-integration-api/src/main/scala/platform/indexer/IndexerConfig.scala:18]: I think he actually added one :-)

**Review by kamil-da — COMMENTED** [review]
Looks good!

A minor thing: we could try to capture CLI option names into constants to avoid hardcoding them in multiple places.
